amcl/src/amcl_node.cpp

```diff
@@ -1511,12 +1654,31 @@ AmclNode::handleInitialPoseMessage(const geometry_msgs::PoseWithCovarianceStampe
 void
 AmclNode::applyInitialPose()
 {
-  boost::recursive_mutex::scoped_lock cfl(configuration_mutex_);
+  std::lock_guard<std::recursive_mutex> cfl(configuration_mutex_);
   if( initial_pose_hyp_ != NULL && map_ != NULL ) {
     pf_init(pf_, initial_pose_hyp_->pf_pose_mean, initial_pose_hyp_->pf_pose_cov);
     pf_init_ = false;
+      ROS_DEBUG("\nInitial pose: %6.3f %6.3f %6.3f",
+               initial_pose_hyp_->pf_pose_mean.v[0],
+               initial_pose_hyp_->pf_pose_mean.v[1],
+               initial_pose_hyp_->pf_pose_mean.v[2]);
+
 
     delete initial_pose_hyp_;
     initial_pose_hyp_ = NULL;
   }
 }
+
+std::string
+AmclNode::stripLeadingSlash(const std::string& frame_name)
+{
+  if (frame_name.size() > 0)
+    if (frame_name[0] == '/')
```

Comment:
It turns out that TF2 removes them for you https://github.com/ros2/geometry2/blob/c5b4d4738ec1f824af364c9e55fb35239de3f1c6/tf2/src/buffer_core.cpp#L228
This was added pretty early in the porting process so it might have been for ROS 1. It seems to work fine without so I removed in 3b804a352c1f1b459710ef285b0dd070737896c1